manifests/repo.pp

```diff
@@ -22,4 +22,14 @@
     enabled  => true,
   }
   -> anchor { 'katello::repo': } # lint:ignore:anchor_resource
+
+  if $facts['os']['release']['major'] == '8' and ($repo_version == 'nightly' or versioncmp($repo_version, '4.3') >= 0) {
+    package { 'katello-dnf-module':
+      ensure      => $dist,
+      name        => 'katello',
+      enable_only => true,
+      provider    => 'dnfmodule',
+      require     => Yumrepo['katello'],
```

Comment:
This probably needs a `before => Anchor['katello::repo']`.

Technical explanation: you can use a collector on a resource (anchor in this case) but not on a class. So in code you can write `Anchor <| title == 'katello::repo' |> -> Package['katello']` and be sure that if `katello::repo` was included, it's done but if it isn't, then it's ignored.